app/sidero-controller-manager/internal/ipxe/ipxe_server.go

```diff
@@ -405,14 +407,21 @@ func newEnvironmentFromServerClass(serverBinding *infrav1.ServerBinding) (env *m
 }
 
 func appendTalosArguments(env *metalv1alpha1.Environment) {
+	if sideroLinkConfig.ServerAddress.IsZero() {
```

Comment:
Maybe can load it in `newEnvironment` call. Then it will be possible to handle an error there.

Reply:
yeah, it's weird. we can't load before mgr is started, as client is not initialized yet.

we don't have a good pre-hook for it. probably we can put it to some `errgroup` in `main`

Reply:
I ended using `mgr.Add` in `main.go`